### PR TITLE
[Typescript] Adding BombCrypto Marketplace adapter

### PR DESCRIPTION
**Pull request by rkreutz**
Added BombCrypto Hero and House market place adapter
- BHero: https://www.bscscan.com/address/0x376A10E7f125A4E0a567cc08043c695Cd8EDd704
- BHouse: https://www.bscscan.com/address/0x049896f350C802CD5C91134E5f35Ec55FA8f0108

### CODE REVIEW

File: adapters/bhero-market/index.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        this.events = ["Sold"];
+        this.pathToAbi = path.join(__dirname, "./abi.json");
+        this.range = 500;
+        this.chunkSize = 6;
+        this.sdk = undefined;
+    }
+
+    run = async () => {
+        this.sdk = this.loadSdk();
+        const token = await this.sdk.callContractMethod('bcoinContract');
```
Comment (rkreutz) [inline adapters/bhero-market/index.js:28]: Notice that even though I've specified the token address on constructor I'm still fetching the token address from the contract, just to be sure

Comment (liutaurasjanuskas) [reply adapters/bhero-market/index.js:28]: Set it to `null`/`undefined` in the constructor then, please.

Comment (rkreutz) [reply adapters/bhero-market/index.js:28]: Done [f59ea13](https://github.com/dappradar/nft-sales-adapters/pull/60/commits/f59ea13933251c3433063dd40f71c57b2c8685d2)

let me know if there is anything else

**Comment by rkreutz** [comment]
@liutaurasjanuskas I've updated the block number for the bomb crypto hero adapter so it points to the block when the contract was created [68d91d2](https://github.com/dappradar/nft-sales-adapters/pull/60/commits/68d91d2f3feff752d7417e8365a561fa77694268)

I've run the tests using the provided command in the README (though I had to make some changes locally to have it call a separate RPC, since I don't have an API key) and it passed. Let me know if you have any issues.

Cheers

File: tester/test.js [hunk 2]
```diff
@@ -30,6 +30,10 @@ const tester = async () => {
     };
     await adapter.run();
 
+    if (array.length == 0) {
+        throw new Error("No entities were returned")
+    }
```
Comment (rkreutz) [inline tester/test.js:35]: I've added an extra check here so there aren't false positives when the array is empty

**Comment by rkreutz** [comment]
@liutaurasjanuskas I've updated the adapters to use typescript

File: src/adapters/bhero-market/index.ts [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+    loadSdk = () => {
+        return new BSC(this);
+    };
+
+    stop = async () => {
+        this.sdk.stop();
+    };
+
+    getBuyer = async (event: EventData): Promise<string | null> => {
```
Comment (liutaurasjanuskas) [inline src/adapters/bhero-market/index.ts:65]: No need for promise here.

File: src/adapters/bhero-market/index.ts [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        return event.returnValues.buyer;
+    };
+
+    getSeller = async (event: EventData): Promise<string | null> => {
```
Comment (liutaurasjanuskas) [inline src/adapters/bhero-market/index.ts:69]: Promise here also doesn't make any sense.

File: src/tester/test.ts [hunk 5]
```diff
@@ -39,6 +39,10 @@ const tester = async (): Promise<void> => {
 
         await adapter.run();
 
+        if (entities.length == 0) {
```
Comment (liutaurasjanuskas) [inline src/tester/test.ts:42]: I don't think you will ever get this error, because it's waiting until at least 4 entities are found before starting tests.

Comment (rkreutz) [reply src/tester/test.ts:42]: I had this issue before migrating to typescript, where I used a block close to the current height and no events were triggered. It had passed even though it had nothing to test, maybe we should leave this just to be cautious?

File: src/adapters/bhouse-market/index.ts [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    getSeller = async (event: EventData): Promise<string | null> => {
+        return event.returnValues.seller;
+    };
+
+    process = async (event: EventData): Promise<ISaleEntity | undefined> => {
+        const block = await this.sdk.getBlock(event.blockNumber);
+        const timestamp = moment.unix(block.timestamp).utc();
+        const po = await priceSdk.get(this.token || '', this.protocol, block.timestamp);
+        const nativePrice = new BigNumber(event.returnValues.price).dividedBy(10 ** (this.symbol?.decimals || 0));
+        const buyer = await this.getBuyer(event);
```
Comment (rkreutz) [inline src/adapters/bhouse-market/index.ts:78]:
```suggestion
        const buyer = this.getBuyer(event);
```

File: src/adapters/bhouse-market/index.ts [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+        if (!buyer) {
+            return;
+        }
+        const tokenId = event.returnValues.tokenId;
+        const seller = await this.getSeller(event);
```
Comment (rkreutz) [inline src/adapters/bhouse-market/index.ts:83]:
```suggestion
        const seller = this.getSeller(event) || "";
```

**Review by rkreutz — COMMENTED** [review]
Suggestions

**Comment by rkreutz** [comment]
@liutaurasjanuskas comments addressed

**Comment by Mintiss** [comment]
Tests were passed

**Pull request merged**